### ISSUE TITLE
`Vonage\Numbers\Client::searchAvailable()` is passing filter as object instead of array to `parseParameters()`

### ISSUE THREAD

**Issue by dragonmantank** [Vonage/vonage-php-sdk-core#249]
## Expected Behavior

```php
$filter = new AvailableNumbers([
    "pattern" => NUMBER_SEARCH_CRITERIA,
    "search_pattern" => (int) NUMBER_SEARCH_PATTERN,
    "type" => VONAGE_NUMBER_TYPE,
    "features" => VONAGE_NUMBER_FEATURES,
]);
$response = $client->numbers()->searchAvailable(COUNTRY_CODE, $filter);
```

should return a list of available numbers.

## Current Behavior

The above code generates the following error:

```
PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Vonage\Numbers\Client::parseParameters() must be of the type array, object given
```

## Possible Solution

`Vonage\Numbers\Client::searchAvailable()` needs to convert the filter data to an array before passing it to `parseParameters()`

## Steps to Reproduce (for bugs)

1. Run the sample code in the description above

## Context

This stops the method from working at all if you use a filter, which is the recommended way. While a raw array will work, the expectation is to use a filter.

## Your Environment

* Version used: v2.3.0
* Environment name and version (e.g. PHP 5.4 on nginx 1.9.1): PHP 7.1

**Comment by prashant-tholia** [comment]
Hi, I'd like to work on this issue.

**Comment by leonardomaier** [comment]
Hey @dragonmantank, can I resolve this issue? 

I was thinking of ways to resolve this. At first I saw that I could implement an AvailableNumbers::toArray() function to return all parameters in an array, but then I noticed that the AvailableNumbers::getQuery() function does just that. 

What do you think about this?

**Comment by leonardomaier** [comment]
Hey @dragonmantank, can I resolve this issue?

I was thinking of ways to resolve this. At first I thought of type hinting `$options` as a `FilterInterface` and then I saw that you said that it is recommended to use a `FilterInterface` but not required, as an array can also be used.

I tried type casting but it didn't work because the object has protected properties causing the array to have the wrong keys.

Maybe we can check if `$options` is an instance of `FilterInterface` and then use `FilterInterface::getQuery()` to get the object converted in array.

What do you think about this?

**Comment by leonardomaier** [comment]
Something like this:

```php
if ($options instanceof FilterInterface) {
    $options = $options->getQuery();
}

**Comment by dragonmantank** [comment]
That should work fine. `getQuery()` would be the easiest way to get an array from a Filter, and is baked into the filter interface. 

Thanks for looking into this!

**Comment by dragonmantank** [comment]
Was fixed with #255
